compiler/src/dotty/tools/dotc/core/ConstraintHandling.scala

```diff
@@ -376,7 +387,12 @@ trait ConstraintHandling {
         else tp
 
       def addParamBound(bound: TypeParamRef) =
-        if (fromBelow) addLess(bound, param) else addLess(param, bound)
+        constraint.entry(param) match {
+          case _: TypeBounds =>
+            if (fromBelow) addLess(bound, param) else addLess(param, bound)
+          case tp =>
+            if (fromBelow) isSubType(bound, tp) else isSubType(tp, bound)
```

Comment:
Added case prompted by a failure. But it looks like we'd need it anyway, and we just failed to have a test that exercises this pass before.